databricks/koalas/frame.py

```diff
@@ -4372,6 +4372,9 @@ def drop(self, labels=None, axis=1,
                               in zip(self._internal.data_columns, self._internal.column_index)
                               if idx not in drop_column_index))
             internal = self._internal.copy(data_columns=list(cols), column_index=list(idx))
```

Comment:
I think you can:

```python
internal = self._internal.copy(
    sdf=self._sdf.drop(*columns),
    data_columns=list(cols),
    column_index=list(idx))
```

You're still modifying immutable `internal` instance.

Reply:
Wonderful! Thanks for your advice and i just fixed them not to modifying immutable internal instance anymore.